### PR TITLE
feat: Add custom headers for import

### PR DESCRIPTION
**Pull request by swetabar**
## Related Issues

[SITES-24434](https://jira.corp.adobe.com/browse/SITES-24434)
We want Import as a service to be able to pass authentication information while trying to import a secured site.

This could be part of the options object (or a new requestBody attribute such as `customHeader`:{})  while making a post request to /tools/import/jobs.

It is currently possible in the helix-importer-ui: (Link: https://github.com/adobe/helix-importer-ui/blob/main/importer-guidelines.md#custom-headers)

We have to consider security - we do not want to store the bearerToken in the DB.

Reference PRs:
https://github.com/adobe/spacecat-import-job-manager/pull/80
https://github.com/adobe/spacecat-content-scraper/pull/145

### CODE REVIEW

**Comment by codevoyage** [comment]
Logging options and headers will be helpful for debugging through the logs. I noticed that logging entries aren’t present in the three PRs. Are there any other types of content, aside from authorization, that we might expect in custom headers?

File: src/support/import-supervisor.js [hunk 1]
```diff
@@ -176,19 +177,36 @@ function ImportSupervisor(services, config) {
...
     // Determine if there is a free import queue
     const importQueueId = await getAvailableImportQueue(importApiKey);
 
     // Hash the API Key to ensure it is not stored in plain text
     const hashedApiKey = hashWithSHA256(importApiKey);
 
+    let updatedOptions = { ...options };
+
+    if (importScript) {
+      updatedOptions = { ...updatedOptions, isCustomJavascriptProvided: true };
```
Comment (blefebvre) [inline src/support/import-supervisor.js:197]: Suggestion for readability:
```suggestion
      updatedOptions = { ...updatedOptions, hasCustomImportJs: true };
```

File: src/support/import-supervisor.js [hunk 2]
```diff
@@ -176,19 +177,36 @@ function ImportSupervisor(services, config) {
...
+    }
+
+    if (customHeaders) {
+      updatedOptions = { ...updatedOptions, isCustomHeadersProvided: true };
```
Comment (blefebvre) [inline src/support/import-supervisor.js:201]:
```suggestion
      updatedOptions = { ...updatedOptions, hasCustomHeaders: true };
```

File: test/controllers/import.test.js [hunk 3]
```diff
@@ -158,6 +158,9 @@ describe('ImportController tests', () => {
           enableJavaScript: 'true',
         },
         importScript: 'QW5kIGV2',
+        customHeaders: {
+          Authorization: 'Bearer aXsPb3183G',
```
Comment (blefebvre) [inline test/controllers/import.test.js:162]: Please add an assertion (or if needed, new test) to verify that the headers are present in the SQS message.

**Review by blefebvre — COMMENTED** [review]
Looks good, and I like that we're not storing the headers in the data layer. Minor items to review.

**Comment by blefebvre** [comment]
Can `validateRequestData(data)` by updated to verify that customHeaders, if provided, is an object?

**Review by blefebvre — APPROVED** [review]

**Pull request merged**